src/node_interface.rs

```diff
@@ -363,81 +267,66 @@ impl NodeInterface {
+        let res_json = self.parse_response_to_json(res)?;
 
-        let res = self.send_post_req(endpoint, body.to_string())?;
+        let mut headers: Vec<Header> = vec![];
 
-        if res.status().is_success() {
-            Ok(true)
-        } else {
-            let json = self.parse_response_to_json(Ok(res))?;
-            Err(NodeError::BadRequest(json["error"].to_string()))
+        for i in 0.. {
+            let header_json = &res_json[i];
+            if header_json.is_null() {
+                break;
+            } else if let Ok(header) = from_str(&header_json.to_string()) {
+                headers.push(header);
+            }
         }
+        Ok(headers)
     }
-}
 
-#[serde_as]
-#[derive(serde::Deserialize, serde::Serialize)]
-pub struct WalletStatus {
-    #[serde(rename = "isInitialized")]
-    pub initialized: bool,
-    #[serde(rename = "isUnlocked")]
-    pub unlocked: bool,
-    #[serde(rename = "changeAddress")]
-    #[serde_as(as = "NoneAsEmptyString")]
-    pub change_address: Option<P2PKAddressString>,
-    #[serde(rename = "walletHeight")]
-    pub height: BlockHeight,
-    #[serde(rename = "error")]
-    pub error: Option<String>,
-}
+    /// Checks if the blockchain indexer is active by querying the node.
+    pub fn indexer_status(&self) -> Result<IndexerStatus> {
+        let endpoint = "/blockchain/indexedHeight";
+        let res = self.send_get_req(endpoint);
+        let res_json = self.parse_response_to_json(res)?;
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn test_parsing_wallet_status_unlocked() {
-        let node_response_json_str = r#"{
-          "isInitialized": true,
-          "isUnlocked": true,
-          "changeAddress": "3Wwc4HWrTcYkRycPNhEUSwNNBdqSBuiHy2zFvjMHukccxE77BaX3",
-          "walletHeight": 251965,
-          "error": ""
-        }"#;
-        let t: WalletStatus = serde_json::from_str(node_response_json_str).unwrap();
-        assert_eq!(t.height, 251965);
-    }
+        let error = res_json["error"].clone();
+        if !error.is_null() {
+            return Ok(IndexerStatus {
+                is_active: false,
+                is_sync: false,
+            });
+        }
 
-    #[test]
-    fn test_parsing_wallet_status_locked() {
-        let node_response_json_str = r#"{
-          "isInitialized": true,
-          "isUnlocked": false,
-          "changeAddress": "",
-          "walletHeight": 251965,
-          "error": ""
-        }"#;
-        let t: WalletStatus = serde_json::from_str(node_response_json_str).unwrap();
-        assert_eq!(t.change_address, None);
-        assert_eq!(t.height, 251965);
+        let full_height = res_json["fullHeight"].as_u64().ok_or(NodeError::FailedParsingNodeResponse(res_json.to_string()))?;
+        let indexed_height = res_json["indexedHeight"].as_u64().ok_or(NodeError::FailedParsingNodeResponse(res_json.to_string()))?;
+
+        let is_sync = (full_height as i64 - indexed_height as i64).abs() < 10;
```

Comment:
nit: Can be changed to full_height.abs_diff(indexed_height) < 10

Reply:
done!